vagrant/src/main/java/org/jclouds/vagrant/compute/VagrantComputeServiceAdapter.java

```diff
@@ -0,0 +1,362 @@
+      this.nodeContainer.mkdirs();
+   }
+
+   private static class MachineName {
+      private String group;
+      private String name;
+      public MachineName(String group, String name) {
+         this.group = group;
+         this.name = name;
+      }
+      public MachineName(String id) {
+         String[] arr = id.split("/");
+         group = arr[0];
+         name = arr[1];
+      }
+      public String getGroup() {
+         return group;
+      }
+      public String getName() {
+         return name;
+      }
+   }
+
+   @Override
+   public NodeAndInitialCredentials<VagrantNode> createNodeWithGroupEncodedIntoName(String group, String name, Template template) {
+      MachineName machineName = new MachineName(group, name);
+      VagrantApi vagrant = getMachine(machineName);
+      init(vagrant.getPath(), name, template);
+
+      Machine newMachine = new Machine();
+      newMachine.setId(group + "/" + name);
+      newMachine.setName(name);
+      newMachine.setStatus(MachineState.POWER_OFF);
+      newMachine.setPath(getMachinePath(machineName));
+
+//      Machine newMachine = vagrant.status(machineName.getName());
+//      newMachine.setId(group + "/" + name);
+
+      VagrantNode node = new VagrantNode(newMachine);
+
+      machines.put(newMachine.getId(), node);
+
+      return startMachine(vagrant, node);
+   }
+
+   private NodeAndInitialCredentials<VagrantNode> startMachine(VagrantApi vagrant, VagrantNode node) {
+      Machine newMachine = node.getMachine();
+
+      vagrant.up(newMachine.getName());
+
+      SshConfig sshConfig = vagrant.sshConfig(newMachine.getName());
+      node.setSshConfig(sshConfig);
+      node.setNetworks(getNetworks(newMachine.getName(), vagrant));
+      node.setHostname(getHostname(newMachine.getName(), vagrant));
+      node.setMachineState(MachineState.RUNNING);
+      newMachine.setStatus(MachineState.RUNNING);
+
+      String privateKey;
+      try {
+         privateKey = Files.toString(new File(sshConfig.getIdentityFile()), Charset.defaultCharset());
+      } catch (IOException e) {
+         throw new IllegalStateException("Invalid private key " + sshConfig.getIdentityFile(), e);
+      }
+
+      LoginCredentials loginCredentials = LoginCredentials.builder()
+            .user(sshConfig.getUser())
+            .privateKey(privateKey)
+            .build();
+      return new NodeAndInitialCredentials<VagrantNode>(node, newMachine.getId(), loginCredentials);
+   }
+
+   private Collection<String> getNetworks(String name, VagrantApi vagrant) {
+       String networks = vagrant.ssh(name, "ip address show | grep 'scope global'");
+       Matcher m = INTERFACE.matcher(networks);
+       Collection<String> ips = new ArrayList<String>();
+       while (m.find()) {
+          String network = m.group(1);
+          // TODO figure out a more generic approach to ignore unreachable networkds (this one is the NAT'd address).
+          if (network.startsWith("10.")) continue;
```

Comment:
Do you mean to do it on the host? I'd be hesitant to execute shell commands on the machine jclouds is running on.